UI/window-dock-youtube-app.cpp

```diff
@@ -0,0 +1,386 @@
+			} else {
+				blog(LOG_ERROR,
+				     "YT: InitYTUserUrl() Failed to get channel id");
+			}
+		}
+	} else
+		blog(LOG_ERROR, "YT: InitYTUserUrl() User is not signed");
+
+	blog(LOG_DEBUG, "YT: InitYTUserUrl() User url: %s", user_url.c_str());
+	return user_url;
+}
+
+void YouTubeAppDock::AddYouTubeAppDock(const QString &title)
+{
+	QString bId(QUuid::createUuid().toString());
+	bId.replace(QRegularExpression("[{}-]"), "");
+	this->setProperty("uuid", bId);
+	this->setObjectName(title + "Object");
+	this->resize(580, 500);
+	this->setMinimumSize(400, 300);
+	this->setWindowTitle(title);
+	this->setAllowedAreas(Qt::AllDockWidgetAreas);
+
+	OBSBasic::Get()->addDockWidget(Qt::RightDockWidgetArea, this);
+	actionYTAppDock = OBSBasic::Get()->AddDockWidget(this);
+
+	if (IsYTServiceSelected()) {
+		const std::string url = InitYTUserUrl();
+		CreateBrowserWidget(url);
+
+		// reload panel layout
+		const char *dockStateStr = config_get_string(
+			App()->GlobalConfig(), "BasicWindow", "DockState");
+		if (dockStateStr) {
+			QByteArray dockState = QByteArray::fromBase64(
+				QByteArray(dockStateStr));
+			OBSBasic::Get()->restoreState(dockState);
+		}
+	} else {
+		this->setVisible(false);
+		actionYTAppDock->setVisible(false);
+	}
+}
+
+void YouTubeAppDock::CreateBrowserWidget(const std::string &url)
+{
+	std::string dir_name = std::string("obs_profile_cookies_youtube/") +
+			       config_get_string(OBSBasic::Get()->Config(),
+						 "Panels", "CookieId");
+	if (cookieManager)
+		delete cookieManager;
+	cookieManager = cef->create_cookie_manager(dir_name, true);
+
+	if (dockBrowser)
+		delete dockBrowser;
+	dockBrowser = cef->create_widget(this, url, cookieManager);
+	if (!dockBrowser)
+		return;
+
+	if (obs_browser_qcef_version() >= 1)
+		dockBrowser->allowAllPopups(true);
+
+	this->SetWidget(dockBrowser);
+	Update();
+}
+
+void YouTubeAppDock::SetVisibleYTAppDockInMenu(bool visible)
+{
+	if (!actionYTAppDock)
+		return;
+
+	actionYTAppDock->setVisible(visible);
+	this->setVisible(visible);
+}
+
+void YouTubeAppDock::Close()
+{
+	this->close();
+}
+
+// only 'ACCOUNT' mode supported
+void YouTubeAppDock::BroadcastCreated(const char *stream_id)
+{
+	DispatchYTEvent(BROADCAST_CREATED, stream_id, YTSM_ACCOUNT);
+}
+
+// only 'ACCOUNT' mode supported
+void YouTubeAppDock::BroadcastSelected(const char *stream_id)
+{
+	DispatchYTEvent(BROADCAST_SELECTED, stream_id, YTSM_ACCOUNT);
+}
+
+// both 'ACCOUNT' and "STREAM_KEY" modes supported
+void YouTubeAppDock::IngestionStarted(const char *stream_id,
+				      streaming_mode_t mode)
+{
+	DispatchYTEvent(INGESTION_STARTED, stream_id, mode);
+}
+
+// both 'ACCOUNT' and "STREAM_KEY" modes supported
+void YouTubeAppDock::IngestionStopped(const char *stream_id,
+				      streaming_mode_t mode)
+{
+	DispatchYTEvent(INGESTION_STOPPED, stream_id, mode);
+}
+
+void YouTubeAppDock::LoggedOut()
+{
+	// LOGGED_OUT;
+}
+
+void YouTubeAppDock::showEvent(QShowEvent *event)
+{
+	(void)event;
```

Comment:
I think you can drop `event` from the function definition here and it would do the same thing.

Reply:
Done.